configure.ac

```diff
@@ -3715,6 +3706,22 @@ AC_CHECK_LIB([dl], [dlopen])       # Dynamic linking for SunOS/Solaris and SYSV
 AC_CHECK_LIB([dld], [shl_load])    # Dynamic linking for HP-UX
 
 
+dnl for faulthandler
+AC_CHECK_HEADERS([execinfo.h link.h dlfcn.h], [
+  AC_CHECK_FUNCS([backtrace dladdr1], [
+    # dladdr1 requires -ldl
+    ac_cv_require_ldl=yes
+  ])
+])
+
+dnl only add -ldl to LDFLAGS if it isn't already part of LIBS (GH-133081)
+AS_VAR_IF([ac_cv_require_ldl], [yes], [
+  AS_VAR_IF([ac_cv_lib_dl_dlopen], [yes], [], [
+    AS_VAR_APPEND([LDFLAGS], [" -ldl"])
```

Comment:
Would autoconf be smart enough if we were to just use AC_CHECK_LIB for dladdr1 as we did for dlopen in that it wouldn't readd the -ldl flag or is it the only way to write these extra checks?

Reply:
Honestly, I don't know - I know enough autoconf to get by, but not much more. 

Unfortunately, I don't have ready access to a development platform that would use the `dladdr1` check (at least, I don't think I do), so it's difficult for me to experiment with this. 

My concern would be whether it would add the flag to `LIBS` or `LDFLAGS` though - if it's added to `LIBS`, we'll be back in the same situation as before. Based on what I'm seeing in other `AC_CHECK_LIB` usage, it looks like it would get added to `LIBS` by default, and any strategy that got it into `LDFLAGS` would essentially be no better than `AS_VAR_APPEND`.

Reply:
I see. Let's keep it that way (it works, it's a bit fragile, but we'll manage if there are issues in the future)